constants.go

```diff
@@ -12,6 +12,7 @@ type Priority int
 
 const severityMask = 0x07
 const facilityMask = 0xf8
+const appNameMax = 48 // limit to 48 chars as per RFC5424
```

Comment:
`appNameMaxChars` or `appNameMaxLength` is more descriptive, but I wonder if this would be better as a local value inside the `RFC5424Formatter` function (because it's RFC5424-specific).